genenetwork/genenetwork2
# Fix unicode rendering errors in show_traits and results page

#### Description
This PR fixes the unicode rendering issues as shown below:

<details>
<summary>Click to view images</summary>

![author](https://user-images.githubusercontent.com/11820306/86187596-d4c3e680-bb44-11ea-84ef-f588d0f0ba12.png)

![ug_show_traits](https://user-images.githubusercontent.com/11820306/86187648-f3c27880-bb44-11ea-8937-930b7a963665.png)

</details>

#### How should be this tested

A server has been set up on: http://gn2.genenetwork.org

Example search: http://gn2.genenetwork.org/search?species=mouse&group=BXD&type=Phenotypes&dataset=BXDPublish&search_terms_or=ppm&search_terms_and=&FormID=searchResult

Only check for text encoding errors

#### Any background context you want to provide

On GN1 things are rendered properly, but on GN2, things are rendered differently yet they use the same db. The biggest difference between the 2 is that they use different libraries to interact with the db. GN1 uses mysqldb, and GN2 uses sqlalchemy. This PR tweaks fetches to the db(using sqlalchemy) to get the raw bytes as they are without doing any unicode transformation, then later in Python, we convert it into unicode. This works because at one point(or maybe still are), we were storing utf-8 bytes into latin1 fields. So the "fix" is the inverse of this process.

#### What are the relevent pivotal issue trackers

https://github.com/genenetwork/genenetwork2/issues/392

#### Screenshots (if appropriate)

N/A

#### Questions

Right now, are the tables still latin1?
